openapi/components/schemas/Common/CommonSubscriptionOrder.yaml

```
@@ -27,16 +27,14 @@ allOf:
         description: >-
           To use plan defaults do not send the `trial` key, or send a `null`.
           value with it.
-        required:
-          - endTime
         properties:
           enabled:
             description: >-
               Enable or disable the trial for this subscription. If enabled for
               plans without trial prices, the trial will be free.
             type: boolean
           endTime:
-            description: The time the trial should end.
+            description: The time the trial should end. Requires a value if enabled.
```

nitpick: It's best to be explicit in descriptions  

```suggestion
            description: Time and date at which the trial ends. If a trial is enabled on this subscription, a value must be provided.
```